volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,690 @@
+                valName.Publisher.value,
+                renderers.UnparsableValue(),
+            )
+            prod_name = file_key.get_string_value(
+                valName.ProductName.value,
+                renderers.UnparsableValue(),
+            )
+            prod_ver = file_key.get_string_value(
+                valName.ProductVersion.value,
+                renderers.UnparsableValue(),
+            )
+
+            program_id = (
+                file_key.get_string_value(
+                    valName.ProgramID.value,
+                    "",
+                )
+                or ""
+            )
+
+            if isinstance(linkdate_str, str):
+                try:
+                    linkdate = datetime.datetime.strptime(
+                        linkdate_str, "%m/%d/%Y %H:%M:%S"
+                    )
+                except ValueError:
+                    linkdate = renderers.UnparsableValue()
+            else:
+                linkdate = linkdate_str
+
+            if isinstance(prod_ver, str):
+                if isinstance(prod_name, str):
+                    prod_name = f"{prod_name} {prod_ver}"
+                else:
+                    prod_name = f"UNKNOWN {prod_ver}"
+
+            yield program_id, AmcacheEntry(
+                AmcacheEntryType.File.name,
+                path=path,
+                company=publisher,
+                last_modify_time=last_mod,
+                compile_time=linkdate,
+                sha1_hash=sha1_hash,
+                product_name=prod_name,
+            )
+
+    @classmethod
+    def parse_driver_binary_key(
+        cls, driver_binary_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[AmcacheEntry]:
+        """Parses information about installed drivers from the `Root\\InventoryDriverBinary` registry key.
+
+        :param driver_binary_key: The `Root\\InventoryDriverBinary` registry key
+        :return: An iterator of `AmcacheEntry`s
+        """
+        for binary_key in driver_binary_key.get_subkeys():
+
+            valName = Win10DriverBinaryValName
+
+            # Depending on the Windows version, the key name will be either the name
+            # of the driver, or its SHA1 hash.
+            if "/" in binary_key.get_name():
+                driver_name = binary_key.get_name()
+                sha1_hash = binary_key.get_string_value(
+                    valName.DriverId.name, renderers.UnparsableValue()
+                )
+            else:
+                sha1_hash = binary_key.get_name()
+                driver_name = binary_key.get_string_value(
+                    valName.DriverName.name,
+                    renderers.UnparsableValue(),
+                )
+
+            if isinstance(sha1_hash, str):
+                sha1_hash = sha1_hash[4:] if sha1_hash.startswith("0000") else sha1_hash
+
+            company, product, service, last_write_time, driver_timestamp = (
+                binary_key.get_string_value(
```

Comment:
This makes for very long-winded code, because we can't use `None` as the default and then change `None` to `UnparseableValue()` using `or UnparseableValue()` even though that's the majority of the cases here (since `None` as the default causes it to throw the exception)...  Not certain what to do here but it feels like having to specify a default value every time is not ideal.

I'm leaning to just returning `None` whenever it can't deal with a value and the people could call `get_value_data` if they needed it to throw the wrong_type exception.  Otherwise the `get_binary_data` methods, etc, are all just glorified ways of catching the exception?

Reply:
That would then allow all the `get_x_data(thing, UnparsableValues())` calls to be swapped out for `or UnparsableValue()` at the time the variable is spat out of the generator.

Reply:
I think I was trying too hard to avoid the requirements of the language (handling exceptions) in order to prevent future runtime errors. It's probably better to go back to the way things were before, for the most part, and plugin authors will just have to be careful about checking values types in their own code.